pkg/cloudprovider/cloudprovider.go

```diff
@@ -87,9 +106,20 @@ type NodeEgressIPConfiguration struct {
 	Capacity  capacity `json:"capacity"`
 }
 
-func NewCloudProviderClient(platformType, platformRegion, secretOverride string) (CloudProviderIntf, error) {
+func NewCloudProviderClient(kubeClient *kubernetes.Clientset, configClient configclientset.Interface, secretOverride string) (CloudProviderIntf, error) {
 	var cloudProviderIntf CloudProviderIntf
 
+	// Get the cloud provider configuration from the API
+	infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", v1.GetOptions{})
```

Comment:
We can't do it like this. There can be no hard dependency from the CNCC on other OpenShift components, this is because upstream OVN-Kubernetes needs to be able to deploy this component and have it work by only providing the secrets required to talk to the cloud API, as mentioned in the enhancement proposal: https://github.com/alexanderConstantinescu/enhancements/blob/3231f1103145f5ef5b7b88bc6aab811b834088b0/enhancements/network/cloud-egress-ip.md#openshift-vs-upstream-project . Upstream should not to deploy other OpenShift components to have this one work. Hence we need a default behavior intended for reading "simple credenentials/secrets" and then override that in case we deploy something "obscure only OpenShift does" type of thing.

Reply:
This is also the reason the `platform-type` / `platform-region` etc is passed to this component though its deployment stanza